lib/networks/networks_list.go

```diff
@@ -0,0 +1,1101 @@
+package networks
+
+// GetAllNetworks returns all supported networks
+func GetAllNetworks() Networks {
+	return all
+}
+
+// GetByChainID returns the network configuration by chain ID
+func GetByChainID(chainID uint64) *Network {
+	for _, network := range all {
+		if network.ChainID == chainID {
+			return network
```

Comment:
`all` is a map, isn't it? :) you can look up the network directly and return `*Network, bool`